### PR TITLE
Bundles up ts typings with dts-bundle

### PR DESCRIPTION
**Pull request by jamonholmgren**
Per discussion in #308, this bundles up the typescript types in one file using [dts-bundle](https://github.com/TypeStrong/dts-bundle). Not sure how good the types are, but it does produce convincing output.

### CODE REVIEW

**Comment by skellock** [comment]
The `.d.ts` files that `tsc` will have relative `import` statements.  That's how they connect all the files together.  Your `types` in `package.json` will point to the top of that tree (often coming out of an `index.ts` file).

I wouldn't sweat chasing a bundled `.d.ts`.  These are dev dependencies.

File: scripts/bundle-d-ts.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var dts = require('dts-bundle')
+
+dts.bundle({
+	name: 'gluegun',
```
Comment (GantMan) [inline scripts/bundle-d-ts.js]: dat space dough

**Comment by GantMan** [comment]
I'm agreeing with Steve.  Unless you can show a significant benefit, I say drop the d.ts bundle.  It's cool that it exists, but I highly doubt you're getting your technical debt's worth.

**Comment by jamonholmgren** [comment]
No problem. This is apparently an area [with a fair amount of disagreement](https://github.com/Microsoft/TypeScript/issues/17196#issuecomment-315397965) over in TypeScript land, so I'll go with the de facto standard here with lots of files. (I like the cleanliness of 1 file, but agree in practice it won't have much of an impact).

**Pull request closed without merging**